util/src/main/java/io/camunda/zeebe/util/sched/ActorTimerQueue.java

```diff
@@ -52,6 +49,13 @@ public void schedule(final TimerSubscription timer, final ActorClock now) {
 
     final long timerId = scheduleTimer(deadline);
     timer.setTimerId(timerId);
+    if (timerJobMap.containsKey(timerId)) {
+      throw new IllegalStateException(
+          "Failed scheduling, timer with id "
+              + timerId
+              + " already exists: "
+              + timerJobMap.get(timerId));
```

Comment:
I would suggest implementing `TimerSubscription#toString()` to gather additional information on the already scheduled timer. WDYT?

Reply:
Makes sense, I added a toString that includes most properties. :+1: